### ISSUE TITLE
questions after reading the paper

### ISSUE THREAD

**Issue by DiTo97** [robot-learning-freiburg/MoMa-LLM#1]
hey there,

At our laboratory we read your paper, but still have a few doubts:

- what models did you use for semantic labeling?
- how fast is the pipeline running in its three components?
- what LLMs did you use for scene graph construction, reasoning and planning?
- could you better explain how the dynamic updating of the scene takes place?

The last part is quite exciting, but the paper is very unclear and vague on how you approached it.

**Comment by martinbchnr** [comment]
Hello @DiTo97,

thanks for your interest. 

- Our approach relies on GT perception as written at the beginning of Section IV. We tested our approach in two distinct settings: First, we use iGibson as a simulator that features a 'semantic camera' that allows us to infer object instances and their semantic class. In the second setting, we operated in a real-world environment. To adapt our system, we constructed a BEV map using the Toyota HSR-mounted 2D LiDAR sensor and labeled objects manually. Whenever the robot uncovers certain areas in the real world, we reveal the underlying semantic BEV map.
- In general, most time is spent on the LLM calls, which can take more than a second. So far, we did not measure this time explicitly.
- GPT-3.5 is used for room classification only while GPT-4 is used for reasoning/planning.
- At each high-level step, we obtain a semantic map and recompute the scene graph based on it. Up to now, we did not tackle incremental scene graph updates. In case, you are referring to dynamic updates of the scene itself, we rely on the physics simulator within iGibson.

**Comment by dHonerkamp** [comment]
Hi,

- we haven't measured the time explicitly, but it does dominate over the scene graph construction. On the other hand, we only require it for high-level reasoning, which happens at a much lower frequency
- we found that GPT3.5 struggles with the reasoning task, and would lead to a large drop in performance. We haven't tested it with other LLMs so far. The room classification task is indeed much simpler, in another project we also found Gemini to produce decent labels. Also, we found that the downstream reasoning is quite robust to imperfect room labels, so the impact of that is not so big.
- correct - we even only recompute it for the high-level reasoning steps, not at each map change. But a more advanced implementation could just run it continuously as a parallel process
